src/OmniSharp.Abstractions/FileWatching/IFileSystemWatcher.cs

```diff
@@ -1,12 +1,20 @@
-﻿using System;
+﻿using OmniSharp.Models.FilesChanged;
+using System;
 
 namespace OmniSharp.FileWatching
 {
     // TODO: Flesh out this API more
     public interface IFileSystemWatcher
     {
-        void Watch(string path, Action<string> callback);
+        void Watch(string path, Action<string, FileChangeType?> callback);
 
-        void TriggerChange(string path);
+        /// <summary>
+        /// Called when a file is created, changed, or deleted.
+        /// </summary>
+        /// <param name="path">The path to the file</param>
+        /// <param name="verb">The type of change. Hosts are not required to pass a change type</param>
+        void TriggerChange(string path, FileChangeType? verb);
```

Comment:
IMO, the terms `type` and `verb` should match. So, if the type is `FileChangeType`, the parameter here should be `type`. Or, the type should be `FileChangeVerb`.

Reply:
it somehow feels that an enum with default value would be more intuitive than nullable enum? at least that's what [MSDN](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/enum) would tell us 😀

Reply:
Fixed names and added a default value.